frame/treasury/src/lib.rs

```diff
@@ -514,14 +674,20 @@ mod tests {
 		new_test_ext().execute_with(|| {
 			// Check that accumulate works when we have Some value in Dummy already.
 			Ring::make_free_balance_be(&Treasury::account_id(), 101);
-			assert_eq!(Treasury::pot(), 100);
+			assert_eq!(Treasury::pot::<Ring>(), 100);
+			// assert_eq!(Treasury::pot::<Kton>(), 0);
```

Comment:
Not sure about why `KTON` and `RING` point to one `Currency`.

Reply:
Solved this with different `storage` names, for example:

### KTON
```rust
decl_storage! {
    trait Store for Module<T: Trait> as KtonBalances { 
        // ...
    }
}
```

### RING
```rust
decl_storage! {
    trait Store for Module<T: Trait> as RingBalances { 
        // ...
    }
}
```